### PR TITLE
AWS S3 event sources v3 client

### LINKED ISSUE #1194: [FEATURE] Support S3 events
**Is your feature request related to a problem? Please describe.**
Like Dropbox/OneDrive, let me kick off events when files are added/updated/deleted from a S3 bucket.

**Describe the solution you'd like**
Pretty much the above. :)

**Do you have a workaround?**
Nope.

### PR DESCRIPTION
**Pull request by andrewjschuang**
Continuation of #1797. Applies the changes and uses AWS v3 clients.

Please review this along with #2302, since it depends on it (this branch is one commit ahead).

Resolves #1797, Resolves #1194.

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/aws/actions/cloudwatch-logs-put-log-event/cloudwatch-logs-put-log-event.mjs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+export default {
+  ...common,
+  key: "aws-cloudwatch-logs-put-log-event",
+  name: "CloudWatch Logs - Put Log Event",
+  description: toSingleLineString(`
+    Uploads a log event to the specified log stream.
+    [See docs](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-cloudwatch-logs/classes/putlogeventscommand.html)
+  `),
+  version: "0.1.0",
```
Comment (jcortes) [inline components/aws/actions/cloudwatch-logs-put-log-event/cloudwatch-logs-put-log-event.mjs:12]: According to a diff with `master` branch you should increment version here and other components as well right?
```suggestion
  version: "0.1.1",
```
Action components like
```
dynamodb-create-table.mjs
dynamodb-execute-statement.mjs
dynamodb-get-item.mjs
dynamodb-put-item.mjs
dynamodb-query.mjs
dynamodb-scan.mjs
dynamodb-update-item.mjs
dynamodb-update-table.mjs
eventbridge-send-event.mjs
lambda-create-function.mjs
lambda-invoke-function.mjs
s3-stream-file.mjs
s3-upload-file.mjs
sns-send-message.mjs
sqs-send-message.mjs
```
abd source components like 
```
new-emails-sent-to-ses-catch-all-domain.mjs
new-records-returned-by-cloudwatch-logs-insights-query.mjs
```

File: components/aws/common/common-cloudwatch-logs.mjs [hunk 2]
```diff
@@ -0,0 +1,118 @@
+import aws from "../aws.app.mjs";
```
Comment (jcortes) [inline components/aws/common/common-cloudwatch-logs.mjs:1]: Optional:

I think is not necessary to have a filename like `common-cloudwatch-logs.mjs` especially if it is inside the `common` dir. My suggestion is to rename these files with just `cloudwatch-logs.mjs` as an example

**Review by jcortes — CHANGES_REQUESTED** [review]
Hi @andrewjschuang everything looks just great to me. I've added just a couple of observations and that's it. Great job!

File: components/aws/actions/cloudwatch-logs-put-log-event/cloudwatch-logs-put-log-event.mjs (hunk 1 above)
Comment (andrewjschuang) [reply components/aws/actions/cloudwatch-logs-put-log-event/cloudwatch-logs-put-log-event.mjs:12]: Right, it's that this PR is on top of #2302, and that one bumps the versions. I'm not sure if right before merging there will be a need to rev, I'm going to check that before merging

File: components/aws/common/common-cloudwatch-logs.mjs (hunk 2 above)
Comment (andrewjschuang) [reply components/aws/common/common-cloudwatch-logs.mjs:1]: Yeah, that sounds good. I'll create another PR to rename all common files after everything is approved and merged!

**Comment by andrewjschuang** [comment]
Hey @jcortes, now that #2302 is merged, I updated this PR.
Could you please check on it real quick? I've commented on your requested changes.

**Review by michelle0927 — APPROVED** [review]
LGTM!

**Comment by DilanAthukorala** [comment]
Verification completed.

**Pull request merged**